src/algorithm/CGAlgorithmsDD.cpp

```diff
@@ -158,6 +158,9 @@ CGAlgorithmsDD::intersection(const CoordinateXY& p1, const CoordinateXY& p2,
 CoordinateXY
 CGAlgorithmsDD::circumcentreDD(const CoordinateXY& a, const CoordinateXY& b, const CoordinateXY& c)
 {
+    if((a == b) && (a == c)){
+        return a;
+    }
```

Comment:
For this case, the actual floating point error happened in `DD::selfDivide`, in case of dividing DD(0) with DD(0) (which is triggered by `numx / denom` below when a, b, and c are identical).  
(I was first thinking to fix it in `DD::selfDivide`, but not directly clear what it then should return, and I assume in general to avoid the FE you need to avoid doing the 0/0 division).

(sidenote: I am not sure what the code below would actually do if only one pair of a, b, and c would be identical)